src/Mollie/WC/Helper/PaymentMethodsIconUrl.php

```diff
@@ -39,7 +62,130 @@ protected function fallToAssets($paymentMethodName)
             return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . 's.svg');
         }
 
-        return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . '.svg');
+        return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . self::SVG_FILE_EXTENSION
+        );
+    }
+
+    /**
+     * Method that handles the composition of the new composed svg file
+     * and returns its url
+     *
+     * @param array $enabledCreditCards Array with the enabled creditcards
+     *
+     * @return string the url to the composed svg
+     */
+    protected function composeSvgImage($enabledCreditCards)
+    {
+        $this->writeToSvgFile();
+
+        $actual = $this->buildSvgComposed($enabledCreditCards);
+
+        return  $this->writeToSvgFile($actual);
+    }
+
+    /**
+     * Is the customization of credit card icons enabled?
+     *
+     * @return bool
+     */
+    protected function enabledCreditcardOptions()
+    {
+        $creditcardSettings = get_option(
+            'mollie_wc_gateway_creditcard_settings'
+        );
+        if (!isset($creditcardSettings['mollie_creditcard_icons_enabler'])) {
+            return false;
+        }
+        return wc_string_to_bool(
+            $creditcardSettings['mollie_creditcard_icons_enabler']
+        );
+    }
+
+    /**
+     * @return array Array containing the credit cards names enabled in settings
+     *               to make customization of checkout icons
+     */
+    protected function enabledCreditcards()
+    {
+        $optionLexem = self::MOLLIE_CREDITCARD_ICONS;
+        $creditcardsAvailable = self::AVAILABLE_CREDITCARD_ICONS;
+        $svgFileName = self::SVG_FILE_EXTENSION;
+        $enabledCreditcards = [];
+
+        $creditcardSettings = get_option('mollie_wc_gateway_creditcard_settings');
+        foreach ($creditcardsAvailable as $card) {
+            if ($creditcardSettings[$optionLexem . $card] === 'yes') {
+                $enabledCreditcards[] = $card . $svgFileName;
+            }
+        }
+
+        return $enabledCreditcards;
+    }
+
+    /**
+     * Method to write the svg file with the composed svg
+     * Can be used to reset the file to empty
+     *
+     * @param string $content svg content to compose the image if empty will
+     *                        reset the file
+     *
+     * @return string the url to the written file
+     */
+    protected function writeToSvgFile($content = '')
+    {
+        $uploads = wp_upload_dir();
+        file_put_contents(
+            $uploads['basedir'] . '/compositeCards.svg',
+            $content
+        );
+        return $uploads['baseurl'] . '/compositeCards.svg';
+    }
```

Comment:
Thinking about this as a generic function that stores something under the base uploads dir, I would pass the filename as a parameter instead of coupling it to the method.

There are two more things to consider here.

1. Why use `file_put_contents` instead of the Wp_Filesystem API?
2. Why this function returns a URL and not the result of the storage? I mean what if `file_put_contents` fails?

About the latter, consider this is a command method, therefore I would let return nothing but throw.
I see you return the URL because you don't want to call again `wp_upload_dir` but you're sacrificing correctness for that little convenience.

And one more thing, when you don't access to `$this` it's convenient to use `static` so PHP does not have to take care of bounding context.